src/store/login/reducers.js

```diff
@@ -0,0 +1,26 @@
+import actions from './actionTypes';
+
+let user = JSON.parse(localStorage.getItem('user'));
+
+const initialState = user ? { loggedIn: true, user } : {};
+
+export default (state = initialState, action) => {
+  switch (action.type) {
+    case actions.LOGIN_REQUEST:
+      return {
+        loggingIn: true,
+        user: action.user,
+      };
+    case actions.LOGIN_SUCCESS:
+      return {
+        loggedIn: true,
+        user: action.user,
+      };
+    case actions.LOGIN_FAILURE:
```

Comment:
Because the results for login_failure and logout are the same you can remove the return statement from login_failure and just have:
`case actions.LOGIN_FAILURE:
case actions.LOGOUT:
return {}`
In the case that two switch cases have the same response, if the first is true (login_failure) it will 'fall through' to the next case and run that response. In this case `return {}`